src/object/cli_obj.c

```diff
@@ -1120,7 +1135,8 @@ obj_shard_tgts_query(struct dc_object *obj, uint32_t map_ver, uint32_t shard,
 	}
 
 	if (rc == 0) {
-		if (obj_op_is_ec_fetch(obj_auxi) && obj_shard->do_rebuilding) {
+		if (obj_op_is_ec_fetch(obj_auxi) && obj_shard->do_rebuilding &&
+		    !obj_auxi->spec_shard) {
```

Comment:
This is a bit confusing, could you please add some comments here?

Reply:
fetch spec_shard only in testing code, for example in ec agg test verify specific shard's ext exist or not, if do deg fetch will break the test.
original code did not set is_ec_obj flag for that case, so did not trigger failure. This patch fix that so expose that failure. here is to fix that case.

Reply:
>fetch spec_shard only in testing code, for example in ec agg test verify specific shard's ext exist or not, if do deg fetch will break the test.

But do_rebuilding should be 0 for specific shard? Sorry these flags becomes very confusing right now, so want to be clarify a bit. Thanks.